### ISSUE TITLE
Error after inserting non-ASCII char from Cliboard to TextInput widget

### ISSUE THREAD

**Issue by macropas** [kivy/kivy#2876]
Xubuntu 14.10, Python 2.7.8.
Inserting non-ASCII char "Д" by Control+V into the TextInput widget

![screenshot - 01222015 - 01 33 58 am](https://cloud.githubusercontent.com/assets/7488502/5847630/e38d4300-a1d7-11e4-8fff-469e69aee2d0.png)

```
[INFO              ] [Logger      ] Record log in /root/.kivy/logs/kivy_15-01-22_0.txt

 [INFO              ] Kivy v1.9.0-dev
 [INFO              ] [Python      ] v2.7.8 (default, Oct 20 2014, 15:05:19) 
 [GCC 4.9.1]
 [INFO              ] [Factory     ] 173 symbols loaded
 [INFO              ] [Image       ] Providers: img_tex, img_dds, img_gif, img_pygame, img_pil (img_ffpyplayer ignored)
 [INFO              ] [Window      ] Provider: pygame(['window_egl_rpi'] ignored)
 [INFO              ] [GL          ] OpenGL version <3.0 Mesa 10.3.2>
 [INFO              ] [GL          ] OpenGL vendor <Intel Open Source Technology Center>
 [INFO              ] [GL          ] OpenGL renderer <Mesa DRI Intel(R) Sandybridge Mobile >
 [INFO              ] [GL          ] OpenGL parsed version: 3, 0
 [INFO              ] [GL          ] Shading version <1.30>
 [INFO              ] [GL          ] Texture max size <8192>
 [INFO              ] [GL          ] Texture max units <16>
 [INFO              ] [Window      ] virtual keyboard not allowed, single mode, not docked
 [INFO              ] [Text        ] Provider: pygame
 [INFO              ] [Video       ] Provider: pygst
 [INFO              ] [OSC         ] using <multiprocessing> for socket
 [INFO              ] [ProbeSysfs  ] device match: /dev/input/event5
 [INFO              ] [MTD         ] Read event from </dev/input/event5>
 [INFO              ] [Base        ] Start application main loop
 [INFO              ] [MTD         ] </dev/input/event5> range position X is 1212 - 5756
 [INFO              ] [MTD         ] </dev/input/event5> range position Y is 996 - 4876
 [INFO              ] [MTD         ] </dev/input/event5> range touch major is 0 - 0
 [INFO              ] [MTD         ] </dev/input/event5> range touch minor is 0 - 0
 [INFO              ] [MTD         ] </dev/input/event5> range pressure is 0 - 255
 [INFO              ] [MTD         ] </dev/input/event5> axes invertion: X is 0, Y is 0
 [INFO              ] [GL          ] NPOT texture support is available
 [INFO              ] [Clipboard   ] Provider: pygame(['clipboard_dbusklipper'] ignored)
 [INFO              ] [Base        ] Leaving application in progress...
 Traceback (most recent call last):
   File "/home/johndoe/kivy-master/examples/RST_Editor/main.py", line 61, in <module>
     Editor().run()
   File "/usr/lib/python2.7/dist-packages/kivy/app.py", line 824, in run
     runTouchApp()
   File "/usr/lib/python2.7/dist-packages/kivy/base.py", line 484, in runTouchApp
     EventLoop.window.mainloop()
   File "/usr/lib/python2.7/dist-packages/kivy/core/window/window_pygame.py", line 381, in mainloop
     self._mainloop()
   File "/usr/lib/python2.7/dist-packages/kivy/core/window/window_pygame.py", line 348, in _mainloop
     self.modifiers):
   File "_event.pyx", line 697, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:6788)
   File "_event.pyx", line 1159, in kivy._event.EventObservers.dispatch (kivy/_event.c:11470)
   File "_event.pyx", line 1083, in kivy._event.EventObservers._dispatch (kivy/_event.c:11066)
   File "/usr/lib/python2.7/dist-packages/kivy/core/window/__init__.py", line 149, in _on_window_key_down
     return self.dispatch('on_key_down', keycode, text, modifiers)
   File "_event.pyx", line 697, in kivy._event.EventDispatcher.dispatch (kivy/_event.c:6788)
   File "_event.pyx", line 1159, in kivy._event.EventObservers.dispatch (kivy/_event.c:11470)
   File "_event.pyx", line 1083, in kivy._event.EventObservers._dispatch (kivy/_event.c:11066)
   File "/usr/lib/python2.7/dist-packages/kivy/uix/textinput.py", line 2017, in keyboard_on_key_down
     self.paste()
   File "/usr/lib/python2.7/dist-packages/kivy/uix/textinput.py", line 1388, in paste
     data = Clipboard.paste()
   File "/usr/lib/python2.7/dist-packages/kivy/core/clipboard/__init__.py", line 88, in paste
     return self._paste()
   File "/usr/lib/python2.7/dist-packages/kivy/core/clipboard/__init__.py", line 106, in _paste
     data = self.get(mime_type)
   File "/usr/lib/python2.7/dist-packages/kivy/core/clipboard/clipboard_pygame.py", line 35, in get
     text = text.encode('utf-8')
 UnicodeDecodeError: 'ascii' codec can't decode byte 0xd0 in position 0: ordinal not in range(128)
```

**Comment by hey-sancho** [comment]
I have the same error.

It looks like `pygame.scrap.get()` returns string, and the following `text.encode('utf-8')` works correctly with unicode string.

The following patch fixed the issue for me:
```
diff --git a/kivy/core/clipboard/clipboard_pygame.py b/kivy/core/clipboard/clipboard_pygame.py
index 6f053ec..8dfbac3 100644
--- a/kivy/core/clipboard/clipboard_pygame.py
+++ b/kivy/core/clipboard/clipboard_pygame.py
@@ -31,8 +31,8 @@ class ClipboardPygame(ClipboardBase):
     def get(self, mimetype='text/plain'):
         self.init()
         text = pygame.scrap.get(mimetype)
-        if PY2:
-            text = text.encode('utf-8')
+        # if PY2:
+        #     text = text.encode('utf-8')
         return text
 
     def put(self, data, mimetype='text/plain'):
```

**Comment by macropas** [comment]
@hey-sancho
It's strange, but there is no any answer from kivy developers...

**Comment by matham** [comment]
I cannot seem to reproduce it on windows with master. Perhaps if it's already `bytes` we should not encode it again.

**Comment by akshayaurora** [comment]
closed via 33bfc526add488571b2dadea7771579a9cbb9042
